core/string/ustring.cpp

```diff
@@ -4002,17 +4002,14 @@ float String::similarity(const String &p_string) const {
 		return 0.0f;
 	}
 
-	Vector<String> src_bigrams = bigrams();
-	Vector<String> tgt_bigrams = p_string.bigrams();
+	const int src_size = length() - 1;
+	const int tgt_size = p_string.length() - 1;
 
-	int src_size = src_bigrams.size();
-	int tgt_size = tgt_bigrams.size();
-
-	int sum = src_size + tgt_size;
+	const int sum = src_size + tgt_size;
 	int inter = 0;
 	for (int i = 0; i < src_size; i++) {
 		for (int j = 0; j < tgt_size; j++) {
-			if (src_bigrams[i] == tgt_bigrams[j]) {
+			if (get(i) == p_string.get(j) && get(i + 1) == p_string.get(j + 1)) {
```

Comment:
Maybe grab the ptr before hand to avoid the `get` calls ? Or put the get(i) and get (i+1) outside the 2nd loop ?

Reply:
Oh right, that would be sensible indeed.